shared/src/commonTest/kotlin/co/nimblehq/blisskmmic/presentation/modules/login/LoginViewModelTest.kt

```diff
@@ -0,0 +1,81 @@
+package co.nimblehq.blisskmmic.presentation.modules.login
+
+import co.nimblehq.blisskmmic.domain.model.Token
+import co.nimblehq.blisskmmic.domain.model.fakeToken
+import co.nimblehq.blisskmmic.domain.repository.MockUserRepository
+import co.nimblehq.blisskmmic.domain.repository.UserRepository
+import co.nimblehq.blisskmmic.domain.usecase.LogInUseCase
+import co.nimblehq.blisskmmic.domain.usecase.LogInUseCaseImpl
+import co.nimblehq.jsonapi.model.JsonApiError
+import co.nimblehq.jsonapi.model.JsonApiException
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.newSingleThreadContext
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.kodein.mock.*
+import org.kodein.mock.tests.TestsWithMocks
+import kotlin.test.*
+
+@ExperimentalCoroutinesApi
+class LoginViewModelTest : TestsWithMocks() {
+
+    private val email = "email@mail.com"
+    private val password = "pass"
+    private val mainThreadSurrogate = newSingleThreadContext("UI thread")
+
+    @Mock
+    lateinit var logInUseCase: LogInUseCase
+    @Fake
+    lateinit var token: Token
+
+    private val loginViewModel by withMocks { LoginViewModel(logInUseCase) }
+
+    override fun setUpMocks() = injectMocks(mocker)
+
+    @BeforeTest
+    fun setUp() {
+        Dispatchers.setMain(mainThreadSurrogate)
+    }
+
+    @AfterTest
+    fun tearDown() {
+        Dispatchers.resetMain()
+        mainThreadSurrogate.close()
+    }
+
+    @Suppress("MaxLineLength")
+    @Test
+    fun `Given LogInUseCase returns success When LogInViewModel calls login Then it should change viewState to success`() = runTest {
```

Comment:
Ditto here about updating to the Android test case naming convention - `<When> _ <Expect> _` instead 🙏